apps/laboratory/tests/shared/pages/ModalPage.ts

```diff
@@ -35,6 +49,32 @@ export class ModalPage {
   }
 
   async load() {
+    if (this.flavor === 'verify-evil') {
+      await this.page.route(`${maliciousUrl}/**/*`, async (route, request) => {
+        let url: string
+        if (request.url() == `${maliciousUrl}/`) {
+          url = `${this.baseURL}/library/verify-evil/`
+        } else {
+          url = request.url().replace(maliciousUrl, this.baseURL)
+        }
+        const response = await fetch(url, {
+          method: request.method(),
+          headers: request.headers(),
+          body: request.postData()
+        })
+        const headers: { [key: string]: string } = {}
+        response.headers.forEach((value, key) => {
+          headers[key] = value
+        })
+        const body = Buffer.from(await response.arrayBuffer())
+        await route.fulfill({
+          status: response.status,
+          headers,
+          body
+        })
+      })
```

Comment:
if not too much trouble I'd move this to some auxiliary method like `routeMaliciousRoute` 
Also, what is it exactly doing? Cause you are routing to the malicious url but replacing the url on the request and proxying the get request?

Reply:
Refactored in latest version